opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,115 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+fn is_metadata(field: &'static str) -> bool {
```

Comment:
this is called for each and every field in every log right? How does this affect the performance? We don't have much benchmarks, but there is stress test which may be leveraged to measure the impact.

Also - do you want to help with writing some logging specific benchmarks?

Reply:
+1 for the benchmark for logs. We need them eventually. I can add them sometime next week unless someone volunteers. For now, good to have metadata validation under the experimental flag, as most of these attributes are not part of the semantic convention either.